odbc/src/api.rs

```diff
@@ -1115,12 +1144,55 @@ pub extern "C" fn SQLSetEnvAttr(
 
 #[no_mangle]
 pub extern "C" fn SQLSetEnvAttrW(
-    _environment_handle: HEnv,
-    _attribute: EnvironmentAttribute,
-    _value: Pointer,
+    environment_handle: HEnv,
+    attribute: EnvironmentAttribute,
+    value: Pointer,
     _string_length: Integer,
 ) -> SqlReturn {
-    unimplemented!()
+    let env_handle = MongoHandleRef::from(environment_handle);
+    env_handle.clear_diagnostics();
+    match env_handle.as_env() {
+        None => SqlReturn::INVALID_HANDLE,
+        Some(env) => match attribute {
+            EnvironmentAttribute::OdbcVersion => match value as Integer {
+                380 => {
+                    let mut env_contents = (*env).write().unwrap();
+                    env_contents.attributes.odbc_ver = value as _;
+                    SqlReturn::SUCCESS
+                }
+                _ => {
+                    env_handle.add_diag_info(ODBCError::InvalidAttrValue("SQL_ATTR_ODBC_VERSION"));
```

Comment:
I don't know if this is compliant. This is for the application to set the version of ODBC it's implemented against. It is a little different from the version that the driver supports which is returned via SqlGetInfo.
I think that the driver is supposed to implement a different behavior based on what the application sets.
But, because I don't want to have to deal with duplicating the error codes (odbc 2 and 3 error codes are different) and slight changes in the odbc state transition. 
If everyone agrees, let's try this and see if users are upset.

Reply:
That makes sense to me. Just updated this part of the function

Reply:
@nbagnard that sounds good to me. Nothing lower than 3.0 though.